src/libmongoc/src/mongoc/mongoc-write-command.c

```diff
@@ -635,6 +653,16 @@ _mongoc_write_opmsg (mongoc_write_command_t *command,
             }
          }
 
+         // If a retry attempt fails with an error labeled NoWritesPerformed,
+         // drivers MUST return the original error.
+         if (mongoc_error_has_label (&reply, "NoWritesPerformed") &&
+             original_error.set) {
```

Comment:
```suggestion
         if (original_error.set &&
             mongoc_error_has_label (&reply, "NoWritesPerformed")) {
```
Switch order to test presence of error before searching for label.